modules/impact.base.entity.d.ts

```diff
@@ -26,26 +26,94 @@ declare global {
         size?: Partial<Vec3>;
         mapId?: number;
       }
+
+      interface Attachable {
+        onEntityKillDetach(this: this): void;
+      }
     }
-    interface Entity extends ig.Class {
+    interface Entity extends ig.Class, ig.Vars.Accessor {
+      id: number;
+      uid: number;
+      mapId: number;
+
       settings: unknown; // an empty object, appears to be unused
       coll: ig.CollEntry;
       isPlayer: boolean | undefined;
+      _killed: boolean;
+      sprites: ig.CubeSprite[];
+      entityAttached: ig.Entity.Attachable[];
+      
+      ballDestroyer?: boolean;
```

Comment:
Misordered properties, also `isPlayer` could get the same treatment as `ballDestroyer`.
```suggestion
      sprites: ig.CubeSprite[];
      entityAttached: ig.Entity.Attachable[];
      _killed: boolean;
      
      ballDestroyer?: boolean;
      isPlayer?: boolean;
```